src/rivian/ble.py

```diff
@@ -0,0 +1,86 @@
+"""Rivian BLE handler."""
+from __future__ import annotations
+import platform
+import asyncio
+import secrets
+import logging
+
+from bleak import BleakClient, BLEDevice
+from rivian import utils
+
+_LOGGER = logging.getLogger(__name__)
+
+PHONE_ID_VEHICLE_ID_UUID = "AA49565A-4D4F-424B-4559-5F5752495445"
+PNONCE_VNONCE_UUID = "E020A15D-E730-4B2C-908B-51DAF9D41E19"
+BLE_ACTIVE_ENTRY_UUID = "5249565F-4D4F-424B-4559-5F5752495445"
+
+NOTIFICATION_TIMEOUT = 3.0
+CONNECT_TIMEOUT = 10.0
+
+async def pair_phone(
+    device: BLEDevice, vehicle_id: str, phone_id: str, vehicle_key: str, private_key: str
+) -> bool:
+    success = False
+
+    # Create an asyncio.Event object to signal the arrival of a new notification.
+    vid_event = asyncio.Event()
+    nonce_event = asyncio.Event()
+    notification_data: bytearray | None = None
+
+    # Callback for notifications from vehicle characteristics 
+    def id_notification_handler(_, data: bytearray) -> None:
+        nonlocal notification_data
+        notification_data = data
+        vid_event.set()
+
+    def nonce_notification_handler(_, data: bytearray) -> None:
+        nonlocal notification_data
+        notification_data = data
+        nonce_event.set()
+
+    # this is a dummy callback (unused)
+    def active_entry_notification_handler(_, data: bytearray) -> None:
+        pass
+
+    try:
+        _LOGGER.debug(f"Connecting to {BLEDevice.name} [{BLEDevice.address}]")
```

Comment:
Use `%s` formatting instead of f-string for logging.

See https://docs.python.org/3/howto/logging.html#optimization